server/service/base_client.go

```diff
@@ -182,3 +191,42 @@ func newBaseClient(
 	}
 	return client, nil
 }
+
+type bodyHandler interface {
+	Handle(*http.Response) error
+}
+
+type FileResponse struct {
+	DestPath     string
+	destFilePath string
+}
+
+func (f *FileResponse) Handle(resp *http.Response) error {
+	_, params, err := mime.ParseMediaType(resp.Header.Get("Content-Disposition"))
+	if err != nil {
+		return fmt.Errorf("parsing media type from response header: %w", err)
+	}
+
+	filename, ok := params["filename"]
+	if !ok {
+		filename = uuid.NewString()
+	}
+
+	f.destFilePath = filepath.Join(f.DestPath, filename)
+	destFile, err := os.Create(f.destFilePath)
+	if err != nil {
+		return fmt.Errorf("creating file: %w", err)
+	}
+	defer destFile.Close()
+
+	_, err = io.Copy(destFile, resp.Body)
+	if err != nil {
+		return fmt.Errorf("copying from http stream to file: %w", err)
+	}
+
```

Comment:
Should still do the explicit `destFile.Close()` here and check the error - it's ok to call Close again in the defer and ignore its error there. It flushes the buffer and confirms the write to disk actually worked (or not).

Reply:
thank you!